src/test/java/seedu/address/logic/commands/GroupCommandTest.java

```diff
@@ -24,62 +31,107 @@
+    public void execute_allFieldsSpecifiedUnfilteredListOnePerson_success() {
+        List<Person> groupedPerson = new ArrayList<>();
+        Set<NusId> nusIds = new HashSet<>();
+        groupedPerson.add(model.getFilteredPersonList().get(0));
+
+        for (Person person : groupedPerson) {
+            nusIds.add(person.getNusId());
+        }
+
+        GroupCommand.GroupPersonDescriptor descriptor = new GroupPersonDescriptorBuilder(groupedPerson).build();
+        GroupCommand groupCommand = new GroupCommand(nusIds, descriptor);
 
         String expectedMessage = String.format(GroupCommand.MESSAGE_GROUP_PERSON_SUCCESS,
                 Messages.format(groupedPerson));
 
         Model expectedModel = new ModelManager(new AddressBook(model.getAddressBook()), new UserPrefs());
-        expectedModel.setPerson(model.getFilteredPersonList().get(0), groupedPerson);
+        expectedModel.setPerson(model.getFilteredPersonList().get(0), groupedPerson.get(0));
+
+        assertCommandSuccess(groupCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Disabled
+    @Test
+    public void execute_allFieldsSpecifiedUnfilteredListMultiplePerson_success() {
+        List<Person> groupedPersons = new ArrayList<>();
+        Set<NusId> nusIds = new HashSet<>();
+        groupedPersons.add(model.getFilteredPersonList().get(0));
+        groupedPersons.add(model.getFilteredPersonList().get(1));
+        for (Person person : groupedPersons) {
+            nusIds.add(person.getNusId());
+        }
+
+        GroupCommand.GroupPersonDescriptor descriptor = new GroupPersonDescriptorBuilder(groupedPersons).build();
+        GroupCommand groupCommand = new GroupCommand(nusIds, descriptor);
+
+        String expectedMessage = String.format(GroupCommand.MESSAGE_GROUP_PERSON_SUCCESS,
+                Messages.format(GroupCommand.createGroupedPerson(groupedPersons, descriptor)));
+
+        Model expectedModel = new ModelManager(new AddressBook(model.getAddressBook()), new UserPrefs());
+
+        IntStream.range(0, groupedPersons.size())
+                .forEach(index -> {
+                    expectedModel.setPerson(model.getFilteredPersonList().get(index), groupedPersons.get(index));
+                });
 
         assertCommandSuccess(groupCommand, model, expectedMessage, expectedModel);
+    }
 
+    @Test
+    public void execute_invalidPersonNusIdFilteredList_failure() {
+        NusId nusId = new NusId("E0000000"); // NusId does not exist
+        Set<NusId> nusIdSet = new HashSet<>();
+        nusIdSet.add(nusId);
+        GroupCommand groupCommand = new GroupCommand(nusIdSet, new GroupCommand.GroupPersonDescriptor());
+        assertCommandFailure(groupCommand, model, Messages.MESSAGE_NON_EXISTENT_PERSON);
     }
 
     @Test
-    public void execute_invalidPersonIndexFilteredList_failure() {
+    public void execute_partialInvalidPersonsNusIdFilteredList_failure() {
```

Comment:
nit: might be nice to explain what this testcase does since "partial invalid" is not immediately clear what the test does